build_dockers.bsh

```diff
@@ -21,13 +21,13 @@ do
   esac
 done
 
-: ${GOLANG_VERSION:=1.23.1}
+: ${GOLANG_VERSION:=1.23.3}
 case "$GOLANG_ARCH" in
   amd64)
-    : ${GOLANG_SHA256:=49bbb517cfa9eee677e1e7897f7cf9cfdbcf49e05f61984a2789136de359f9bd}
+    : ${GOLANG_SHA256:=a0afb9744c00648bafb1b90b4aba5bdb86f424f02f9275399ce0c20b93a2c3a8}
```

Comment:
What is the authoritative source for these hashes?

Reply:
The `fetch_hashes()` [function](https://github.com/git-lfs/build-dockers/blob/45b04563eba7b2083bc447c34d71f1cca2d40bd5/update-hashes#L63-L70) in the `update-hashes` script in this repo just [retrieves](https://github.com/git-lfs/build-dockers/blob/45b04563eba7b2083bc447c34d71f1cca2d40bd5/update-hashes#L64) the JSON-formatted data on the `go.dev` release list [page](https://go.dev/dl/?mode=json) and parses that.  So the authoritative source is the Go project itself.

Reply:
Thanks! I wonder if there would be any value in pinning the `https://go.dev` cert or would that be just annoying to maintain and practically useless?

Reply:
I'm open to a PR which does that, but I confess I may not develop one myself.

Our build process involves a number of steps beyond just these Docker images, and for Windows and macOS we don't use these images at all, nor for the release binaries we build for Linux that we don't package in RPM or Debian packages.

In those cases we depend on the GitHub `actions/setup-go` action to install Go, and it retrieves the Go binaries from cached assets prepared by the `actions/go-versions` action, which as far as I can tell also just [fetches](https://github.com/actions/go-versions/blob/8b69c18c83105cb4b041451d093a16b362c6b988/.github/workflows/get-go-versions.yml#L12) the list of Go versions from `https://go.dev`.

If you think there's real value in tightening up this chain, though, we can definitely discuss that further.